File: 20-valid-parentheses/valid-parentheses.js
var isValid = function(s) {
    let stack = [];
    let top = "";
    let close = "";
    for(let i = 0;i<s.length;i++){
        if (stack.length === 0){
            stack.push(s[i]);
            top = s[i];
            continue;
        }
        if (top === "{")
            close = "}";
        else if (top === "[")
            close = "]";
        else if (top == "(")
            close = ")";
        else
        close = ""
        if (close === s[i]){
            stack.pop();
            top = stack[stack.length-1];
        }
        else{
            stack.push(s[i]);
            top = s[i];
    }
    }
    return stack.length === 0;
};

File: 20-valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for each in s:
            if len(stack) == 0:
                stack.append(each)
                continue
            else:
                if each == ")" and stack[-1] == "(":
                    stack.pop()
                elif each == "}" and stack[-1] == "{":
                    stack.pop()
                elif each == "]" and stack[-1] == "[":
                    stack.pop()
                else:
                    stack.append(each)
        return len(stack) == 0



        